File: lesson2/DemoWebApplication/Components/WeatherForecastComponent/Services/DemoWeatherForecastService.cs
using DemoWebApplication.Components.WeatherForecastComponent.Dtos;
using DemoWebApplication.Components.WeatherForecastComponent.Services.Interfaces;

namespace DemoWebApplication.Components.WeatherForecastComponent.Services;

public class DemoWeatherForecastService : IWeatherForecastService
{
    public WeatherForecastDto GetByLocation(string location)
    {
        return new WeatherForecastDto()
        {
            Location = "Kyiv",
            Humidity = 1,
            Pressure = 1,
            TempMax = 1,
            TempMin = 1
        };
    }

    public async Task<WeatherForecastDto> GetByLocationAsync(string location)
    {
        return GetByLocation(location);
    }
}

File: lesson2/DemoNestJsApplication/src/components/weather-forecast-component/services/demo-weather-forecast.service.ts
import { WeatherForecastDto } from "../dtos/weather-forecast.dto";
import { IWeatherForecastService } from "./interfaces/i.weather-forecast.service";

export class DemoWeatherForecastService implements IWeatherForecastService {
    getByLocation(location: string): WeatherForecastDto {
        return {
            location: "Kyiv",
            humidity: 1,
            pressure: 1,
            tempMax: 1,
            tempMin: 1
        };
    }

    async getByLocationAsync(location: string): Promise<WeatherForecastDto> {
        return this.getByLocation(location);
    }
}